File: 2789. Largest Element in an Array after Merge Operations/Solution.cpp
class Solution {
public:
    long long maxArrayValue(vector<int>& nums) {
        int n = nums.size();
        long long high = 0, curr = 0;
        for (int i = n - 1; i >= 0; i--) {
            curr += nums[i];
            if (i == 0 || nums[i] < nums[i - 1]) {
                high = max(curr, high);
                if (i > 0 && curr < nums[i - 1]) curr = 0;
            }
        }
        return high;
    }
};

File: 2789. Largest Element in an Array after Merge Operations/Solution.py
class Solution:
    def maxArrayValue(self, nums: List[int]) -> int:
        n, high, curr = len(nums), 0, 0
        for i in range(n - 1, -1, -1):
            curr += nums[i]
            if i == 0 or nums[i] < nums[i - 1]:
                high = max(curr, high)
                if i > 0 and curr < nums[i - 1]: curr = 0
        return high
